src/param.rs

```diff
@@ -626,24 +625,42 @@ impl AudioParamProcessor {
     }
 
     fn mix_to_output(&mut self, input: &AudioRenderQuantum, output: &mut AudioRenderQuantum) {
-        #[cfg(test)]
-        assert_eq!(self.buffer.len(), RENDER_QUANTUM_SIZE);
+        if self.buffer.len() == 1 && input.is_silent() {
+            let mut value = self.buffer[0];
 
-        let output = output.channel_data_mut(0);
-        output.copy_from_slice(self.buffer.as_slice());
+            if value.is_nan() {
+                value = self.default_value;
+            }
 
-        output
-            .iter_mut()
-            .zip(input.channel_data(0).iter())
-            .for_each(|(o, i)| {
-                *o += i;
+            output.set_is_single_valued(true);
 
-                if o.is_nan() {
-                    *o = self.default_value;
-                }
+            let output_channel = output.channel_data_mut(0);
+            output_channel[0] = value.clamp(self.min_value, self.max_value);
+        } else {
+            // the param buffer is constant but it should be modulated by an
+            // incoming signal, so we need to resize it to RENDER_QUANTUM_SIZE
```

Comment:
Maybe in this branch it is now simpler to set `output = input` and then use the loop to add in the `buffer`, using the `cycle` trick again. You don't need the resize in that case.

You could also optimize for the case that buffer is a single value of `0.` maybe, then you can skip the whole loop

Reply:
yeah right more simple indeed

> You could also optimize for the case that buffer is a single value of 0. maybe, then you can skip the whole loop

Hum not really, we still need to clamp because we can't trust the incoming signal boundaries. Another case is when `input.is_silent()` then we can `output.copy_from_slice(buffer). In all cases we will need the loop to check `nan`s and clamp, don't know if it worth it. I added a comment in the code, we can see that later I think